jupyterhub/auth.py

```diff
@@ -356,6 +356,9 @@ def is_admin(self, handler, authentication):
                 persisted; and `admin`, the admin setting value for the user.
         """
 
+    def pre_exec_fn(self, user, spawner):
+        """Hook called before exec is called in the spawned child process """
```

Comment:
...when using a LocalProcessSpawner and its derivatives.

btw, this should probably return a pure callable rather than being an instance method calling the method. I wouldn't trust e.g. accessing the database-connection-having user object in the pre_exec_fn.

Instead, let's have something like:

```python
@staticmethod
def _preexec_fn(name, service, encoding):
    ...

def make_preexec_fn(self, user, spawner):
    return partial(self._preexec_fn, user.name, self.service, self.encoding)
```

Reply:
The instance method no longer calls this method, so this should be resolved.